web/packages/shared/components/Controls/MultiselectMenu.tsx

```diff
@@ -31,15 +31,17 @@ import { CheckboxInput } from 'design/Checkbox';
 import { ChevronDown } from 'design/Icon';
 import { HoverTooltip } from 'design/Tooltip';
 
-type MultiselectMenuProps<T> = {
-  options: {
-    value: T;
-    label: string | ReactNode;
-    disabled?: boolean;
-    disabledTooltip?: string;
-  }[];
-  selected: T[];
-  onChange: (selected: T[]) => void;
+type Option<T> = {
+  value: T;
+  label: string | ReactNode;
+  disabled?: boolean;
+  disabledTooltip?: string;
+};
+
+type MultiselectMenuProps<T extends readonly Option<any>[]> = {
+  options: T;
+  selected: Extract<T[number], Option<any>>['value'][];
```

Comment:
I'm curious: why doesn't `T[]` work here? And why don't we parametrize with `T extends Option<any>>` instead? Or even parametrize just with the value type, as it used to be?

Reply:
Using `T extends readonly Option<any>[]` lets us infer the full union of `value`s from the `options` array, so `selected` and `onChange` are correctly typed. Typing just a single `Option<T>` or using just `T[]` as was there previously results in an array of only the `value` from the first `option`